### PR TITLE
pythonPackages.fipy: 3.4.1 -> 3.4.2.1

### PR DESCRIPTION
**Pull request by wd15**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wd15** [comment]
Result of `nixpkgs-review pr 116567` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.fipy</li>
    <li>python39Packages.fipy</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 116567` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>python38Packages.fipy</li>
<li>python39Packages.fipy</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
python38Packages.fipy:
<details>


warning: duplicate-check-inputs
Dependencies listed in `propagatedBuildInputs` should not be repeated in `checkInputs`.

Detected duplicates:

- gmsh-4.8.0

Near pkgs/development/python-modules/fipy/default.nix:45:5:

```
   |
45 |     checkInputs = not_darwin_inputs;
   |     ^
```

Near pkgs/development/python-modules/fipy/default.nix:33:5:

```
   |
33 |     propagatedBuildInputs = [
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/duplicate-check-inputs.md
warning: unused-argument
Unused argument: `fetchPypi`.
Near pkgs/development/python-modules/fipy/default.nix:3:3:

```
  |
3 | , fetchPypi
  |   ^
```
</details>
</details>

File: pkgs/development/python-modules/fipy/default.nix [hunk 1]
```diff
@@ -42,8 +41,6 @@ in
       openssh
     ] ++ lib.optionals isPy27 [ pysparse ] ++ not_darwin_inputs;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/fipy/default.nix:42]:
```suggestion
    ] ++ lib.optionals isPy27 [ pysparse ]
      ++ lib.optionals !stdenv.isDarwin [ gmsh ];
```

File: pkgs/development/python-modules/fipy/default.nix [hunk 2]
```diff
@@ -23,11 +22,11 @@ let
 in
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/fipy/default.nix:22]: and now this can be removed.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/fipy/default.nix (hunk 1 above)
Comment (wd15) [reply pkgs/development/python-modules/fipy/default.nix]: done

File: pkgs/development/python-modules/fipy/default.nix (hunk 2 above)
Comment (wd15) [reply pkgs/development/python-modules/fipy/default.nix]: done

**Comment by wd15** [comment]
This should be ready to merge now.

File: pkgs/development/python-modules/fipy/default.nix [hunk 3]
```diff
@@ -18,41 +17,37 @@
...
+    scipy
+    pyamg
+    matplotlib
+    tkinter
+    mpi4py
+    future
+    scikit-fmm
+    openssh
+  ] ++ lib.optionals isPy27 [ pysparse ]
+    ++ lib.optionals (! stdenv.isDarwin) [ gmsh ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/fipy/default.nix:40]:
```suggestion
  ++ lib.optionals (!stdenv.isDarwin) [ gmsh ];
```

**Pull request merged**